pyglet/graphics/shader.py

```diff
@@ -236,31 +236,128 @@ def __repr__(self):
         return f"Attribute(name='{self.name}', location={self.location}, count={self.count})"
 
 
+class _UniformArray:
+    """Wrapper of the GLSL array data inside a Uniform.
+    Allows access to get and set items for a more Pythonic implementation.
+    Types with a length longer than 1 will be returned as tuples as an inner list would not support individual value
+    reassignment. Array data must either be set in full, or by indexing."""
+
+    _slots_ = ('uniform', 'gl_type', 'gl_getter', 'gl_setter', 'is_matrix', 'dsa', '_c_array', '_ptr')
+    def __init__(self, uniform, gl_getter, gl_setter, gl_type, is_matrix, dsa):
+        self._uniform = uniform
+        self._gl_type = gl_type
+        self._gl_getter = gl_getter
+        self._gl_setter = gl_setter
+        self._is_matrix = is_matrix
+        self._dsa = dsa
+
+        if self._uniform.length > 1:
+            self._c_array = (gl_type * self._uniform.length * self._uniform.size)()
+        else:
+            self._c_array = (gl_type * self._uniform.size)()
+
+        self._ptr = cast(self._c_array, POINTER(gl_type))
+
+    def __len__(self):
+        return self._uniform.size
+
+    def __delitem__(self, key):
+        raise ShaderException("Deleting items is not support for UniformArrays.")
+
+    def __getitem__(self, key):
+        """Return as a tuple.
+        Returning as a list may imply setting inner list elements will update values.
+        """
+        if isinstance(key, slice):
+            start, stop, step = key.start, key.stop, key.step
+            sliced_data = self._c_array[start:stop:step]
+            if self._uniform.length > 1:
+                return [tuple(data) for data in sliced_data]
+            else:
+                return tuple([data for data in sliced_data])
+
+        value = self._c_array[key]
+        return tuple(value) if self._uniform.length > 1 else value
+
+    def __setitem__(self, key, value):
+        if isinstance(key, slice):
+            start, stop, step = key.start, key.stop, key.step
+            self._c_array[start:stop:step] = value
```

Comment:
I could be wrong, but I _think_ we can just pass the slice in directly (instead of splitting into start/stop/step). IE:
```py
            self._c_array[key] = value
```

Reply:
I just tested, this seems to be the case. Nice!